daphne/src/roles/leader.rs

```diff
@@ -439,8 +443,13 @@ pub trait DapLeader<S>: DapAuthorizedSender<S> + DapAggregator<S> {
             .try_put_agg_share_span(task_id, task_config, agg_span)
             .await
             .into_iter()
-            .map(|(_bucket, (result, _report_metadata))| {
-                result.map(|replayed_reports| replayed_reports.len())
+            .map(|(_bucket, (result, _report_metadata))| match result {
+                Ok(()) => Ok(0),
+                Err(MergeAggShareError::AlreadyCollected) => {
+                    panic!("aggregated to a collected agg share")
```

Comment:
Why panic? Wouldn't we want to handle this similarly to detected replays, i.e., just output a log line with the warning?